### PR TITLE
Performance: Load shared frontend plugin dependencies on demand

### PR DESCRIPTION
**Pull request by jackw**
**What is this feature?**

This PR changes the loading strategy of shared plugin dependencies from "up front" to "on demand".

**Why do we need this feature?**

TLDR; To reduce frontend entry-point bundle sizes. 

Grafana shares some frontend NPM packages with plugins. This is done by importing all these packages in a single file and then passing the contents of those modules to systemjs / fe sandbox. This causes all the shared dependencies to be bundled and served up front regardless of what the Grafana user is actually doing / what plugins are loaded. This PR changes the behaviour so a shared dependency will only load when either Grafana itself or a plugin requires it.

_Webpack build stats for entrypoint_

| | Before | After |
| --- | --- | --- |
| Unminified | 26.8 MiB | 21.3 MiB | 
| Minified | 11 MiB| 8.88MiB |

**Who is this feature for?**

Everybody

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**
I have given this a basic test with SystemJS and the Sandbox enabled and plugins continue to load. Careful attention should be given to the changes in the sharedDependencies map to make sure no dependencies were deleted and that the same `useDefault` flags are set on certain dependencies.

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by mckn — APPROVED** [review]
Wow, this looks great!

**Review by academo — APPROVED** [review]
Changes look good.

tested with the sandbox on with some plugins and all work fine.

**Comment by jackw** [comment]
The e2e tests are failing due to this branch making changes to the load order of plugin dependencies which in turn makes an issue with using the older ui extensions apis more apparent. 😞 

We'll need to reach out to plugin developers to understand how to circumvent this before we can merge this PR.

**Comment by jackw** [comment]
FYI - I've updated this branch to fix the playwright plugin extensions tests but we should NOT merge this PR until plugins platform has reached out to our cloud app teams and helped them migrate their apps over to use the latest extension APIs.

**Review by sunker — APPROVED** [review]
LGTM

**Pull request merged**
